kbase/narrative
# DATAUP-312 update appmanager tests, first step toward deduplication

# Description of PR purpose/changes

This is PR 1 of ~3 for adding the `run_app_bulk` method to `biokbase.narrative.jobs.appmanager`.

There were some... issues with that old module. 
1. There's a fair amount of duplicated code. This takes the first step of taking the wrappers around each app running style and turning that into a decorator. Now, instead of `run_app` calling `_run_app_internal`, where `run_app` would just catch and reformat errors for Narrative consumption, this uses a single decorator that applies to each function. It still follows the idiom of sending errors as comm messages to the front end as well as printing a prettier form of error message (which would go under the cell). I applied this only to `run_app` and `run_app_batch` as there are no tests for `run_local_app` right now, and I don't want to make any changes without any regression tests in place. Next PR.
2. Many of the tests for that module just did a basic exercise of the code, and checked to see if they didn't return anything. They also used mocks in the wrong order, and incompletely. This addresses that for the existing tests for `run_app` and `run_app_batch` (none for `run_local_app` yet) by capturing both `stdout` and the mocked calls to the comm channel and verifying them. Found a couple bugs that way, too. Funny what you find when you make tests work.
3. I put in a stub for the `run_app_bulk` function and a couple stubby tests. I'm ok with removing those for now if they're a distraction from this PR and adding them to the next if that makes more sense.

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-312

# Testing Instructions
* Details for how to test the PR:
* Run tests with `make test-backend`.
* Should be no other changes.

# Dev Checklist:
